samples/boards/tlsr9x/factorydata/src/factory_data_parser.c

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright (c) 2025 Telink Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "factory_data_parser.h"
+
+#include <zcbor_decode.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/flash.h>
+
+#include <ctype.h>
+#include <string.h>
+
+#define MAX_FACTORY_DATA_NESTING_LEVEL 3
+
+const struct device *mFlashDevice = DEVICE_DT_GET(DT_CHOSEN(zephyr_flash_controller));
+
+__attribute__((section(
```

Comment:
What is the reason to keep RAM image?
As I understand it's enough to declare structure. And make const pointer to flash area.